libvips/colour/CMYK2XYZ.c

```diff
@@ -150,20 +150,57 @@ vips_CMYK2XYZ_build( VipsObject *object )
 	return( 0 );
 }
 
+#else
+
+typedef VipsColourCode VipsCMYK2XYZ;
+typedef VipsColourCodeClass VipsCMYK2XYZClass;
+
+G_DEFINE_TYPE(VipsCMYK2XYZ, vips_CMYK2XYZ, VIPS_TYPE_COLOUR_CODE);
+
+#define SCALE (VIPS_D65_Y0)
+
+void
+vips_CMYK2XYZ_line( VipsColour *colour, VipsPel *out, VipsPel **in, int width )
+{
+    unsigned char *p = (unsigned char *) in[0];
+    float *q = (float *) out;
+
+    int i;
+
+    for (i = 0; i < width; i++) {
+        float r = 255.0 - ((1.0 / 255.0) * p[0] * (255.0 - p[3]) + p[3]);
+        float g = 255.0 - ((1.0 / 255.0) * p[1] * (255.0 - p[3]) + p[3]);
+        float b = 255.0 - ((1.0 / 255.0) * p[2] * (255.0 - p[3]) + p[3]);
+        
+        float x, y, z;
+        
+        vips_col_scRGB2XYZ(r / 255.0, g / 255.0, b / 255.0, &x, &y, &z);
```

Comment:
This will be rather slow, and I think it's not necessary. You can just set:

```C
x = VIPS_D65_X0 * r / 255.0;
```

etc.

Reply:
I guess you could do all the scaling in a separate step:

```C
float c = p[0] / 255.0;
...
float r = 1.0 - (c * (1.0 - k) + k);
...
q[0] = VIPS_D65_X0 * r;
```

Perhaps that's a bit clearer? Maybe not worth it.